src/nnsight/contexts/GraphBasedContext.py

```diff
@@ -68,17 +80,18 @@ def __exit__(self, exc_type, exc_val, exc_tb) -> None:
 
     ### BACKENDS ########
 
-    def local_backend_execute(self) -> Graph:
-
-        self.graph.execute()
-
-        graph = self.graph
-        graph.alive = False
+    def local_backend_execute(self) -> None:
 
-        if not isinstance(graph, weakref.ProxyType):
-            self.graph = weakref.proxy(graph)
+        try:
+            self.graph.execute()
+        except protocols.EarlyStopProtocol.EarlyStopException as e:
+            raise e
+        finally:
```

Comment:
Does finally actually get called here? Because you are raising a new exception in except.

Reply:
Yes, it's being called!